onelogin/get.go

```diff
@@ -216,3 +199,52 @@ func Get(app, provider string) (*awsprovider.Credentials, error) {
 
 	return &creds, nil
 }
+
+// getDevice returns the MFA device used by the user. If there is
+// more than one available the user is prompted which one should
+// be used.
+func getDevice(devices []Device) (deviceID, deviceType string, err error) {
+	if len(devices) == 0 {
+		// this should never happen
+		err = errors.New("No MFA device returned by Onelogin")
+		return
+	}
+	if len(devices) == 1 {
+		deviceID = fmt.Sprintf("%v", devices[0].DeviceId)
```

Comment:
We want to return a pointer to a `Device` here, like so:

```go
if len(devices) == 1 {
    d = &Device{DeviceID: devices[0].DeviceID, DeviceType: devices[0].DeviceType}
    return
}
```

Of course we should edit the function signature, too (see previous comments).